### PR TITLE
Add tests for Smart Wallet custom logic

### PR DESCRIPTION
**Pull request by mauricioirace**
This PR aims to test basic functionality of smart wallet's custom logic.
It also fixes an issue regarding the address of the logic not being correctly loaded from storage.

### CODE REVIEW

File: contracts/smartwallet/CustomSmartWallet.sol
Comment (mauricioirace) [inline contracts/smartwallet/CustomSmartWallet.sol:304]: This change was needed in order to run the custom logic. Without it, the address retrieved from storage was corrupted, maybe due to a dangling pointer or so.

File: contracts/smartwallet/CustomSmartWallet.sol
Comment (mauricioirace) [inline contracts/smartwallet/CustomSmartWallet.sol:311]: same as above

File: package.json
Comment (mauricioirace) [inline package.json:40]: I should remove this.

File: rsknode/Dockerfile
Comment (mauricioirace) [inline rsknode/Dockerfile:12]: I needed this due to debugging purpose

File: test/TestUtils.ts
Comment (mauricioirace) [inline test/TestUtils.ts:360]: Will remove this

File: test/smartwallet/CustomSmartWallet.test.ts
Comment (mauricioirace) [inline test/smartwallet/CustomSmartWallet.test.ts:64]: I think we could improve this, making a constructor-like function, instead of sharing the request across the context

File: scripts/execute-test
Comment (lucasvuotto) [inline scripts/execute-test:8]: This runs only the first test. Correct idiom would be

```suggestion
if [ $# -eq 0 ]; then
	echo "You need to specify at least one test"
	exit 1
fi
```

...and see the `npx` block below for how to complete it.

File: scripts/execute-test
Comment (lucasvuotto) [inline scripts/execute-test:11]: Please don't mix `bash`-specific syntax with portable shell syntax. Either use `[ "${NETWORK}" = "" ]` (single `=`) or `[[ "${NETWORK}" == "" ]]` (double `[[` and `]]`). I prefer using `[ -z "$NETWORK" ]`, which is the idiomatic way to ask if a string is empty in `sh`.

File: scripts/execute-test
Comment (lucasvuotto) [inline scripts/execute-test:17]:
```suggestion
for test; do
    npx truffle test --show-events --network testing "$test"
done
```

(hand indented as `Tab` doesn't input tabs in here)

File: rsknode/Dockerfile
Comment (lucasvuotto) [inline rsknode/Dockerfile:37]: Can you explain the reason for this changes?

File: run-tests.sh
Comment (lucasvuotto) [inline run-tests.sh:1]: I don't see changes other than using `docker-compose`. I'm fine with dropping the script, but I'd like to understand the reason. Additionally, keep in mind that this does a couple of things that the rewrite does not; in particular it waits for the node to be able to reply before starting the tests.

File: rsknode/Dockerfile
Comment (mauricioirace) [reply rsknode/Dockerfile:37]: It didn't work, as it could not find the directory

Also, the version 2.1.0 didn't show the error messages, I I neeeded them fot debugging purpouses. But if you prefer i can rollback that change and set it to 2.1.0 again

**Review by lucasvuotto — APPROVED** [review]
Well, it seems I completed a review and forgot to submit. In there, the important part was regarding the RSK version: it's `2.1.0` because it was what was in use when I made the script. I highly suggest to keep `2.2.0` or, even better, try out Iris' release candidate.

I'll give my approve just because I'm pending as a reviewer. Otherwise, having no changes related to Docker or scripts, I don't think it adds anything. Can't judge on the rest of the contents of this PR.

File: truffle.js
Comment (JONAF2103) [inline truffle.js:30]: Be careful here

File: test/smartwallet/CustomSmartWallet.test.ts
Comment (mortelli) [inline test/smartwallet/CustomSmartWallet.test.ts:74]: is there a source for this code?

**Review by JONAF2103 — APPROVED** [review]
LGTM, i run the tests and everything is ok, don't forge to remove the extra changes related to docker and scripts that doesn't apply to this PR

File: test/smartwallet/CustomSmartWallet.test.ts
Comment (mortelli) [inline test/smartwallet/CustomSmartWallet.test.ts:186]: sorry, could you please explain what `if logic revert` means?

**Review by mortelli — COMMENTED** [review]
some minor comments below; please remember to also run the linter on this branch.

it's good that we have several tests, i would maybe look into extracting the common code between them to make them a bit more readable.

to be frank though, i feel a bit out of my depth when reading this new `CustomSmartWallet` code. 

having said that, i am willing to give it a deeper look at a later time (hopefully one in which i am a bit more educated on this :sweat_smile:). this could be both before or after merging this pr

File: test/smartwallet/CustomSmartWallet.test.ts
Comment (mauricioirace) [reply test/smartwallet/CustomSmartWallet.test.ts:186]: if the logic fails/reverts the transaction

File: test/smartwallet/CustomSmartWallet.test.ts
Comment (mauricioirace) [reply test/smartwallet/CustomSmartWallet.test.ts:74]: It was repeated in multiple tests. Check SmartWallet.test.ts

File: contracts/interfaces/WalletCustomLogic.sol
Comment (mortelli) [reply contracts/interfaces/WalletCustomLogic.sol]: thanks!

File: test/smartwallet/CustomSmartWallet.test.ts
Comment (mortelli) [reply test/smartwallet/CustomSmartWallet.test.ts:74]: ok, we can extract it later then. not the point of this PR :+1:

File: test/smartwallet/CustomSmartWallet.test.ts
Comment (mortelli) [reply test/smartwallet/CustomSmartWallet.test.ts:186]: thanks!

File: test/smartwallet/CustomSmartWallet.test.ts
Comment (raullaprida) [inline test/smartwallet/CustomSmartWallet.test.ts:173]: Can you read the event from the logs of the transaction receipt? (you can call web3.eth.getTransactionReceipt(txHash) again, as I noticed sometimes the rawLogs from the receipt sent by Truffle does not contain all the logs). Im worried you could be reading an event emitted in a previous transaction

File: test/smartwallet/CustomSmartWallet.test.ts
Comment (raullaprida) [inline test/smartwallet/CustomSmartWallet.test.ts:326]: Same here

**Review by raullaprida — COMMENTED** [review]
I only left a few comments.
Overall I think it looks great, but you may want to add some tests for the CustomSmartWalletFactory

**Pull request merged**
